File: Interview Preparation Kit - CPP/01. Warm-up Challenges/003. Jumping on the Clouds.cpp
// Problem: https://www.hackerrank.com/challenges/jumping-on-the-clouds/problem
// Score: 20


#include <iostream>
#include <vector>
using namespace std;

int main(){
    int n;
    cin >> n;

    int element;
    vector<int> arr;

    while (cin >> element){
        arr.push_back(element);
    }

    int ans = 0;
    int position = 0;

    while (position < n - 1){
        position ++;
        if (position + 1 <= n - 1 && arr[position + 1] != 1){
            position ++;
        }
        ans ++;
    }

    cout << ans;

    return 0;
}

File: Interview Preparation Kit - Python/01. Warm-up Challenges/003. Jumping on the Clouds.py
# Problem: https://www.hackerrank.com/challenges/jumping-on-the-clouds/problem
#  Score: 20


def jumping_on_clouds(n, c):
    c.append(0)
    ans = 0
    position = 0
    while position < n-1:
        position += (c[position+2] == 0) + 1
        ans += 1
    return ans


n = int(input())
c = list(map(int, input().rstrip().split()))
print(jumping_on_clouds(n, c))
